pdns/recursordist/contrib/powerdns-example-script.lua

```diff
@@ -3,140 +3,141 @@ pdnslog("pdns-recursor Lua script starting!", pdns.loglevels.Warning)
 
 malwareset = newDS()
 malwareset:add("nl")
 
 magic2 = newDN("www.magic2.com")
 
-
 magicMetric = getMetric("magic")
 
+badips = newNMG()
+badips:addMask("127.1.0.0/16")
+
+-- this check is applied before any packet parsing is done
+function ipfilter(rem, loc, dh)
+  pdnslog("ipfilter called, rem: "..rem:toStringWithPort().."loc: "..loc:toStringWithPort().." match:"..tostring(badips:match(rem)))
+  pdnslog("id: "..dh:getID().."aa: "..tostring(dh:getAA()).."ad: "..tostring(dh:getAD()).." arcount: "..dh:getARCOUNT())
+  pdnslog("ports: "..rem:getPort().." "..loc:getPort())
+  return badips:match(rem)
+end
+
 -- shows the various ways of blocking, dropping, changing questions
 -- return false to say you did not take over the question, but we'll still listen to 'variable'
 -- to selectively disable the cache
 function preresolve(dq)
-	pdnslog("Got question for "..dq.qname:toString().." from "..dq.remoteaddr:toString().." to "..dq.localaddr:toString())
-
-        local ednssubnet=dq:getEDNSSubnet()
-	if(ednssubnet) then
-			pdnslog("Packet EDNS subnet source: "..ednssubnet:toString()..", "..ednssubnet:getNetwork():toString())
-        end
-
-
-	local a=dq:getEDNSOption(3)
-	if(a) then
-		pdnslog("There is an EDNS option 3 present: "..a)
-	end
-
-	loc = newCA("127.0.0.1")
-	if(dq.remoteaddr:equal(loc))
-	then
-		pdnslog("Query from loopback")
-	end
-
-	-- note that the comparisons below are CaSe InSensiTivE and you don't have to worry about trailing dots
-	if(dq.qname:equal("magic.com"))
-	then
-		magicMetric:inc()
-		pdnslog("Magic!")
-	else
-		pdnslog("not magic..")
-	end
-
-	if(dq.qname:__eq(magic2)) -- we hope to improve this syntax
-	then
-		pdnslog("Faster magic") -- compares against existing DNSName
-	end                           -- sadly, dq.qname == magic2 won't work yet
-
-        if blockset:check(dq.qname) then
-                dq.variable = true  -- disable packet cache in any case
-                if dq.qtype == pdns.A then
-	        	dq:addAnswer(pdns.A, "1.2.3.4")
-        		dq:addAnswer(pdns.TXT, "\"Hello!\"", 3601) -- ttl
-        		return true;
-        	end
-        end
-
-        if dropset:check(dq.qname) then
-        	dq.rcode = pdns.DROP
-        	return true;
-        end
-
-
-
-        if malwareset:check(dq.qname) then
-		dq:addAnswer(pdns.CNAME, "xs.powerdns.com.")
-        	dq.rcode = 0
-        	dq.followupFunction="followCNAMERecords"    -- this makes PowerDNS lookup your CNAME
-        	return true;
-        end
-
-	return false;
+  pdnslog("Got question for "..dq.qname:toString().." from "..dq.remoteaddr:toString().." to "..dq.localaddr:toString())
+
+  local ednssubnet = dq:getEDNSSubnet()
+  if ednssubnet
+  then
+    pdnslog("Packet EDNS subnet source: "..ednssubnet:toString()..", "..ednssubnet:getNetwork():toString())
+  end
+
+  local a = dq:getEDNSOption(3)
+  if a
+  then
+    pdnslog("There is an EDNS option 3 present: "..a)
+  end
+
+  loc = newCA("127.0.0.1")
+  if dq.remoteaddr:equal(loc) then
```

Comment:
If we really want to be consistent with the `then` being on a new line, which I am really not a fan of:
```suggestion
  if dq.remoteaddr:equal(loc)
  then
```